rfcs/0007-properties-file-encoding.md

```diff
@@ -0,0 +1,94 @@
+# RFC 0007 Properties File Encoding
+
+- Start Date: 2019-07-15
+- RFC PR: [#168](https://github.com/SAP/ui5-tooling/pull/168)
+- Issue: [#161](https://github.com/SAP/ui5-tooling/issues/161)
+- Affected components
+  - [x] [ui5-builder](https://github.com/SAP/ui5-builder)
+  - [x] [ui5-server](https://github.com/SAP/ui5-server)
+  - [ ] [ui5-cli](https://github.com/SAP/ui5-cli)
+  - [ ] [ui5-fs](https://github.com/SAP/ui5-fs)
+  - [ ] [ui5-project](https://github.com/SAP/ui5-project)
+  - [ ] [ui5-logger](https://github.com/SAP/ui5-logger)
+
+## Summary
+
+Properties files (`*.properties`) should be encoded in pure ASCII when serving them via the `ui5-server` or building an application/library with the `ui5-builder`.
+
+## Motivation
+
+Currently the properties files are mostly encoded in ISO-8859-1 (which is used by most existing SAP server platforms).
+By default the files are served as UTF-8 by the `ui5-server`. This will lead to the problem that special characters are not displayed correctly.
+The user wants to be able to use properties files with ISO-8859-1 encoding. Additionally the user wants have the option to specify the encoding.
+
+## Detailed design
+
+### Configuration
+
+A configuration for the `ui5-builder` tasks and the `ui5-server` should be provided such that the source encoding can be specified.
+
+There should be 2 options:
+
+- `ISO-8859-1` (non ASCII characters will be escaped with the unicode sequence `\uXXXX`)
+- `UTF-8` (no character escaping will take place)
```

Comment:
> `UTF-8` (no character escaping will take place)

If a server treats the resulting file as being `ISO-8859-1` encoded, we get misinterpreted characters again I suppose.

Don't we need to escape all non-ASCII characters in the UTF-8 case as well? It seems that this is also how https://github.com/SAP/ui5-builder/pull/293 is currently implemented.

Reply:
Also see Matthias' comment [here](https://github.com/SAP/ui5-builder/pull/293/files/6504c6efced1d365e39b213ab6e5b94e40234e99#diff-02d343d30f4acdc0de962fcab14736a2)

Reply:
done